libtenzir/builtins/aggregation-functions/collect.cpp

```diff
@@ -47,13 +49,44 @@ class collect_function final : public aggregation_function {
   }
 
   auto finish() && -> caf::expected<data> override {
+    // XXX: Why exchange? Is this same instance reused?
     return data{std::exchange(result_, {})};
   }
 
   list result_ = {};
 };
 
-class plugin : public virtual aggregation_function_plugin {
+class collect_instance final : public aggregation_instance {
+public:
+  explicit collect_instance(ast::expression expr) : expr_{std::move(expr)} {
+  }
+
+  auto update(const table_slice& input, session ctx) -> void override {
+    auto arg = eval(expr_, input, ctx);
+    if (caf::holds_alternative<null_type>(arg.type)) {
+      return;
+    }
+    // XXX: What should happen if it receives different types?
+    // Currently, everything ends up coerced to strings.
+    for (auto i = int64_t{}; i < arg.array->length(); ++i) {
```

Comment:
This is just a personal suggestion, but I personally always try to be explicit about the value when I'm relying on it: `int64_t{0}`.